model/src/main/kotlin/VcsInfo.kt

```diff
@@ -128,8 +121,9 @@ private class VcsInfoDeserializer : StdDeserializer<VcsInfo>(VcsInfo::class.java
         return VcsInfo(
             VcsType(node["type"].textValueOrEmpty()),
             node["url"].textValueOrEmpty(),
-            node["revision"].textValueOrEmpty(),
-            (node["resolved_revision"] ?: node["resolvedRevision"])?.textValue(),
+            // For backward compatibility, if resolved_revision is set, prefer it over revision because it is more
+            // specific.
+            node["resolved_revision"]?.textValue() ?: node["revision"].textValueOrEmpty(),
```

Comment:
This favors an empty `resolvedRevision` over a non-empty `revision`, intended? ...Same for RepositoryProvenance deserializer.

Reply:
Note that `resolvedRevision` is / was nullable (unlike `revision`). So if there was no `resolvedRevision` is should have been *null* instead of empty. That's why I believe it's highly unlikely to get an empty `resolvedRevision` here.